rosplan_knowledge_base/src/KnowledgeComparitor.cpp

```diff
@@ -138,7 +138,10 @@ namespace KCL_rosplan {
 		case rosplan_knowledge_msgs::KnowledgeItem::FUNCTION:
 
 			// check function value
-			if(a.function_value != b.function_value) return false;
+			// Note: ContainsKnowledge checks if the two KnowledgeItems are the same. For functions, value is NOT
+			//       checked as this may be used to compare. In order to compare the vlaue of the function the
+            //       KnowledgeComparitor::inequalityTrue
+			//if(a.function_value != b.function_value) return false;
 
```

Comment:
Suggested to remove extra comments and commented-out code in order to keep the file clean. Also trying out the new suggestions feature.